Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1393,6 +1394,90 @@ public extension Purchases {
         await self.paywallEventsManager?.track(featureEvent: paywallEvent)
     }
 
+#if ENABLE_AD_EVENTS_TRACKING
+
+    /**
+     Tracks when an ad impression is displayed.
+
+     Call this method from your ad SDK's impression callback to report ad displays to RevenueCat.
+     This enables RevenueCat to track ad impressions alongside your subscription revenue.
+
+     - Parameter data: The displayed ad event data
+
+     ## Example:
+     ```swift
+     await Purchases.shared.trackAdDisplayed(.init(
+         networkName: "AdMob",
+         mediatorName: .appLovin,
+         placement: "home_screen",
+         adUnitId: "ca-app-pub-123",
+         adInstanceId: "instance-456"
+     ))
+     ```
+     */
+    func trackAdDisplayed(_ data: AdDisplayed) async {
```

Comment:
Would it make sense to create some `AdTracker` class that has all these methods? Mostly to avoid polluting the Purchases API with too many methods.

Reply:
I would think the first. Better to keep it as part of the purchases instance.